O/OpenVAF/build_tarballs.jl

```diff
@@ -0,0 +1,56 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+using BinaryBuilderBase: get_addable_spec
+
+name = "OpenVAF"
+version = v"23.5.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/pascalkuthe/OpenVAF.git", "a9697ae7780518f021f9f64e819b3a57033bd39f"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/OpenVAF
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+export OPENVAF_LLVM_LIBNAMES="LLVM"
+if [[ "${target}" == *-mingw* ]]; then
+    export OPENVAF_LLVM_LIBNAMES="LLVM-15jl"
+fi
+
+LLVM_LINK_SHARED=1 LLVM_CONFIG="${host_prefix}/tools/llvm-config" OPENVAF_LLVM_LIBDIR="${libdir}" OPENVAF_LLVM_INCLUDEDIR="${includedir}" cargo build --release --bin openvaf
+
+install -Dvm 0755 "target/${rust_target}/release/openvaf${exeext}" "${bindir}/openvaf${exeext}"
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+filter!(p -> arch(p) ∉ ("powerpc64le", "armv6l", "armv7l"), platforms)
+filter!(p -> !(Sys.islinux(p) && libc(p) == "musl" && arch(p) == "i686"), platforms) # fails to include "llvm/IR/Instructions.h"
+filter!(p -> !(Sys.iswindows(p) && arch(p) == "i686"), platforms) # undefined reference to `_Unwind_Resume'
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("openvaf", :openvaf),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(get_addable_spec("LLVM_full_jll", v"15.0.7+9")),
```

Comment:
If you're doing static linking do you still need to install `LLVM_full_jll` as dependency at runtime?

Reply:
Actually
```console
% otool -L openvaf
openvaf:
	@rpath/libLLVM.dylib (compatibility version 1.0.0, current version 15.0.7)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.3.1)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
	@rpath/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```
this doesn't seem to be statically linking llvm?

Reply:
Yeah, I had to switch this away from static linking due to the `llvm-config` woes. The host LLVM build has a different set of components and static libraries, but it cannot run the target llvm-config so it's quite hard to get a list of the necessary static libs

My workaround for now was to dynamically link, which only has a single libLLVM to worry about.

Probably a more proper fix for the future would be to recompile LLVM fresh in this recipe, since the additional components are needed for OpenVAF to do support cross-compilation between x86 and AArch64 anyway

Reply:
Ok, I _think_ this should work as long as LLVM_full_jll@v15.0.7 can be installed, but you also need to specify the compat bound with something like (untested!)
```suggestion
    Dependency(get_addable_spec("LLVM_full_jll", v"15.0.7+9"); compat="15.0.7"),
```